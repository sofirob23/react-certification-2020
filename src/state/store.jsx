import React, { createContext, useReducer } from 'react';

const myStorage = window.sessionStorage;
const initialState = {
  videoList: [],
  darkMode: false,
  currentVideo: {},
  currentVideoProfile: {},
  loggedUser: myStorage.getItem('loggedUser')
    ? JSON.parse(myStorage.getItem('loggedUser'))
    : null,
  favorites: [],
};
const store = createContext(initialState);
const { Provider } = store;

const StateProvider = ({ children }) => {
  const [state, dispatch] = useReducer((currentState, action) => {
    switch (action.type) {
      case 'search':
        return {
          videoList: action.payload,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites,
        };
      case 'darkMode':
        return {
          videoList: currentState.videoList,
          darkMode: action.payload,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites,
        };
      case 'play':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: action.video,
          currentVideoProfile: action.profile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites,
        };
      case 'login':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: action.payload,
          favorites: currentState.favorites,
        };
      case 'logout':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: null,
          favorites: currentState.favorites,
        };
      case 'addFavorite':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites.append(action.payload),
        };
      case 'removeFavorite':
        return {
          videoList: currentState.videoList,
          darkMode: currentState.darkMode,
          currentVideo: currentState.currentVideo,
          currentVideoProfile: currentState.currentVideoProfile,
          loggedUser: currentState.loggedUser,
          favorites: currentState.favorites.filter(
            (video) => video.id.videoId !== action.payload.id.videoId
          ),
        };
      default:
        throw new Error();
    }
  }, initialState);

  return <Provider value={{ state, dispatch }}>{children}</Provider>;
};

export { store, StateProvider };
