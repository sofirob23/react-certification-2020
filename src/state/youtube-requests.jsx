import axios from 'axios';

const search = {
  part: 'snippet',
  key: process.env.REACT_APP_API_KEY,
};

const youtubeSearch = (requestParams) => {
  const youtube = axios.create({
    baseURL: 'https://www.googleapis.com/youtube/v3/',
    params: Object.assign(search, requestParams),
  });
  return youtube.get('/search');
};

export default youtubeSearch;
