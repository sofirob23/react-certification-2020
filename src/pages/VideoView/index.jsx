import React, { useContext, useEffect } from 'react';
import Grid from '@material-ui/core/Grid';
import VideoPlayer from '../../components/VideoPlayer';
import VideoList from '../../components/VideoList';
import { store } from '../../state/store';

const VideoView = (props) => {
  const globalState = useContext(store);

  useEffect(() => {
    if (globalState.state.darkMode) {
      document.body.style.backgroundColor = 'var(--darkgrey)';
    } else {
      document.body.style.backgroundColor = 'white';
    }
  }, [globalState]);

  return (
    <>
      <Grid container spacing={3}>
        <Grid item xs={12} md={6} lg={7}>
          <VideoPlayer {...props} />
        </Grid>
        <Grid item xs={12} md={6} lg={5}>
          <VideoList {...props} />
        </Grid>
      </Grid>
    </>
  );
};

export default VideoView;
