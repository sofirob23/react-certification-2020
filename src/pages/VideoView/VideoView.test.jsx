import React from 'react';
import renderer from 'react-test-renderer';
import { store } from '../../state/store';
import VideoView from './index';

describe('VideoView Tests', () => {
  const { Provider } = store;
  const dispatch = jest.fn();
  const state = {
    videoList: [],
    darkMode: false,
    currentVideo: {},
    currentVideoProfile: {},
  };
  const props = {
    params: {
      videoID: 'HYyRZiwBWc8',
    },
  };
  test('VideoView renders correctly', () => {
    const tree = renderer
      .create(
        <Provider value={{ dispatch, state }}>
          <VideoView match={props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
