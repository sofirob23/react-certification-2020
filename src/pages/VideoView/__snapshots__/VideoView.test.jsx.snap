// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoView Tests VideoView renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-7"
  >
    <iframe
      allow="accelerometer; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      className="sc-bdfBwQ fxNuLZ"
      data-testid="video"
      frameBorder="0"
      src="https://www.youtube.com/embed/HYyRZiwBWc8?controls=0"
    />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <h1
        className="sc-hKgILt hLfBYY"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          className="sc-dlfnbm hbCwZl"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
          <span
            className="sc-gsTCUz hplYmp"
            data-testid="test-span"
          >
            Add to Favorites
          </span>
        </div>
      </div>
    </div>
    <h1
      className="sc-eCssSg dtwrMX"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-5"
  >
    <div
      className="sc-crrsfI kPiTeu"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      />
    </div>
  </div>
</div>
`;
